Description: Makes the header of the files generated by XJC reproducible
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/jaxb-ri/xjc/src/main/java/com/sun/tools/xjc/Options.java
+++ b/jaxb-ri/xjc/src/main/java/com/sun/tools/xjc/Options.java
@@ -983,10 +983,11 @@
                 + "' "
                 + Messages.format(Messages.TIME_FORMAT);
         SimpleDateFormat dateFormat = new SimpleDateFormat(format, Locale.ENGLISH);
+        dateFormat.setTimeZone(java.util.TimeZone.getTimeZone("UTC"));
 
         return Messages.format(
             Messages.FILE_PROLOG_COMMENT,
-            dateFormat.format(new Date()));
+            dateFormat.format(System.getenv("SOURCE_DATE_EPOCH") != null ? new Date(1000 * Long.valueOf(System.getenv("SOURCE_DATE_EPOCH"))) : new Date()));
     }
 
     /**
